src/PrestaShopBundle/Form/Admin/Sell/Attachment/AttachmentType.php

```diff
@@ -0,0 +1,135 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Sell\Attachment;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\CleanHtml;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\DefaultLanguage;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Config\AttachmentValidationConfiguration;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Attachment form type definition
+ */
+class AttachmentType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('name', TranslatableType::class, [
+                'type' => TextType::class,
+                'required' => true,
+                'options' => [
+                    'constraints' => [
+                        new TypedRegex(
+                            [
+                                'type' => 'generic_name',
+                            ]
+                        ),
+                    ],
+                    new Length(
+                        [
+                            'max' => AttachmentValidationConfiguration::MAX_NAME_LENGTH,
+                            'maxMessage' => $this->trans(
+                                'This field cannot be longer than %limit% characters',
+                                ['%limit%' => AttachmentValidationConfiguration::MAX_NAME_LENGTH],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]
+                    ),
+                ],
+                'constraints' => [
+                    new DefaultLanguage(),
+                ],
+            ])
+            ->add('file_description', TranslatableType::class, [
+                'type' => TextType::class,
+                'required' => false,
+                'options' => [
+                    'constraints' => [
+                        new CleanHtml(),
+                    ],
+                ],
+            ])
+            ->add('file', FileType::class, [
+                'required' => !$options['is_edit_form'] ||
```

Comment:
can you make this easier to read ? for exemple using an intermediate variable name `$isFileRequired`